solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -245,116 +259,92 @@ library DKG {
+                        misbehavedMembersIndices[i],
                     "Corrupted misbehaved members indices"
                 );
             }
+            uint8 last = misbehavedMembersIndices[
+                misbehavedMembersIndices.length - 1
+            ];
+            require(
+                last >= 1 && last <= groupSize,
+                "Corrupted misbehaved members indices"
+            );
         }
 
+        // Each signature needs to be 65 bytes long and signatures need to be
+        // provided.
         uint256 signaturesCount = result.signatures.length / 65;
-        require(result.signatures.length >= 65, "Too short signatures array");
+        require(result.signatures.length > 0, "No signatures provided");
         require(
             result.signatures.length % 65 == 0,
             "Malformed signatures array"
         );
+
+        // We expect the same amount of signatures as the number of declared
+        // group member indices that signed the result.
+        uint256[] calldata signingMembersIndices = result.signingMembersIndices;
         require(
-            signaturesCount == result.signingMembersIndices.length,
+            signaturesCount == signingMembersIndices.length,
             "Unexpected signatures count"
         );
         require(signaturesCount >= groupThreshold, "Too few signatures");
         require(signaturesCount <= groupSize, "Too many signatures");
 
-        bytes32 resultHash = keccak256(
-            abi.encodePacked(
-                result.groupPubKey,
-                result.misbehavedMembersIndices,
-                self.startBlock
-            )
-        );
-
-        bytes memory current; // Current signature to be checked.
-        bool[] memory usedMemberIndices = new bool[](groupSize);
-        // TODO: Revisit at optimization stage. We probably don't need to fetch
-        //       addresses of all members. Addresses of signing members should
-        //       be enough though this will require some additional processing
-        //       here. Compare two approaches gas-wise against a real pool.
-        address[] memory membersAddresses = sortitionPool.getIDOperators(
-            result.members
-        );
-
-        for (uint256 i = 0; i < signaturesCount; i++) {
-            uint256 memberIndex = result.signingMembersIndices[i];
-            require(memberIndex > 0, "Invalid index");
-            require(memberIndex <= result.members.length, "Index out of range");
-
+        // Signing member indices needs to be unique, between [1,64], and sorted
+        // in ascending order.
+        for (uint256 i = 1; i < signingMembersIndices.length; i++) {
             require(
-                !usedMemberIndices[memberIndex - 1],
-                "Duplicate member index"
+                signingMembersIndices[i - 1] >= 1 &&
+                    signingMembersIndices[i - 1] <= groupSize &&
+                    signingMembersIndices[i - 1] < signingMembersIndices[i],
```

Comment:
I think we don't need to check that every single element in `singningMembersIndices` is `>=1` and `<=groupSize`.
Checking that `signingMembersIndices[0] >= 1` and signingMembersIndices[signingMembersIndices.length-1]<= groupSize` should be enough, because we also ensure that  `signingMembersIndices` are sorted.

Reply:
Very good catch!